discovery_engine/lib/src/domain/engine/mock_engine.dart

```diff
@@ -140,4 +140,18 @@ class MockEngine implements Engine {
   ) async {
     _incrementCount('userReacted');
   }
+
+  @override
+  Future<List<DocumentWithActiveData>> activeSearch(
+    String query,
+    int page,
+    int pageSize,
+  ) async {
+    _incrementCount('activeSearch');
+
+    return [
+      DocumentWithActiveData(doc0, active0),
+      DocumentWithActiveData(doc1, active1),
```

Comment:
on the rust side we use the nil stack id for the documents of an active search, we probably have to do the same here? this would mean that we shouldn't reuse `doc0` and `doc1` since they have a non-nil stack id.

Reply:
ah yes thanks, this totally escaped me..

Reply:
to keep it simple it just returns empty for now.
I think returning some documents with nil id could be done separately to this pr.